client-extensions/liferay-object-collection-filter-extension/src/common/components/FilterSettings.js

```diff
@@ -1,54 +1,36 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
 import React, { useState, useEffect } from 'react';
 import Select from 'react-select';
+import ModalComponent from './ModalComponent';
 
 const FilterSettings = ({ initialSettings, onSaveAndClose }) => {
 
     const [settings, setSettings] = useState(initialSettings);
 
-    const [objects, setObjects] = useState([]);
-    const [objectFields, setObjectFields] = useState([]);
-
-    useEffect(() => {
-        // Simulate fetching data (replace with actual fetching logic)
-        setObjects([
-           { label: 'students', value: 'students'},
-           { label: 'teachers', value: 'teachers'},
-       ]);
-        setObjectFields([
-            { label: 'age', fieldName: 'age', fieldType: 'numeric' },
-            { label: 'gender', fieldName: 'gender', fieldType: 'boolean' },
-        ]);
-    }, []);
+    // const handleChange = (field, value) => {
+    //     setSettings((prevSettings) => ({
+    //         ...prevSettings,
+    //         [field]: value,
+    //     }));
+    // };
 
-    const handleChange = (field, value) => {
-        setSettings((prevSettings) => ({
-            ...prevSettings,
-            [field]: value,
-        }));
-    };
+    // const handleClose = () => {
+    //     onSaveAndClose(settings);
+    // };
 
-    const handleClose = () => {
-        onSaveAndClose(settings);
-    };
+    // const handleSaveAndClose = (returnedObject) => {
+    //     console.log(returnedObject);
+    //     onSaveAndClose(settings);
+    //   };
+    console.log(settings);
 
     return (
         <div className="settings-popup">
-            {/* TODO: Refactor. This is just an example.*/}
-            <h2>Settings</h2>
-            <Select
-                options={objects}
-                value={settings.selectedObject}
-                onChange={(object) => handleChange('selectedObject', object.value)}
-            />
-            {settings.selectedObject && (
-                <Select
-                    options={objectFields}
-                    isMulti
-                    value={settings.selectedFields}
-                    onChange={(fields) => handleChange('selectedFields', fields)}
-                />
-            )}
-            <button onClick={handleClose}>Save</button>
+            <ModalComponent onSave={setSettings}/>
```

Comment:
I don't think we can change the settings state in index.js and that might prevent data retrieval for rendering FilterForm. Please check it.

Reply:
Fix it. Thanks